docs/function-buffer-varints.md

```diff
@@ -0,0 +1,72 @@
+
+# Varint Functions for Buffers
+
+## Summary
+
+This RFC proposes the addition of new functions to the buffer API:
+
+- `buffer.readleb128` 
+- `buffer.writeleb128`
+- `buffer.readuleb128`
+- `buffer.writeuleb128` 
+
+These functions will write and read (U)LEB-128 variable-width integers (varints) from a buffer.
+
+## Motivation
+
+One of the top use cases for buffers is compressing and serializing data as small as possible. Numbers consume 8 bytes in their raw form but can be serialized in smaller sizes using buffers to reduce size. However, writing code to serialize as the smallest size can be tedious and confusing for less-experienced programmers.  
+
+The following are implementations of ULEB-128 reading/writing in Luau:
+
+```lua
+local function writeuleb128(stream, value)
```

Comment:
Please update the implementation examples to match what the proposed functions will do with a library implementation.
Right now they don't compile, incorrectly use the `buffer` library functions and don't return the proposed results.
It would be nice to also convert the argument value to integer number using `bit32` library.

Reply:
@bmcq-0 You can update your read implementation with this, adjust accordingly.

```lua
	local function readByte()
		local byte = buffer_readu8(stream, position)
		position = position + 1
		return byte
	end

	local function readVarInt()
		local result = 0

		for i = 0, 4 do
			local value = readByte()
			result = bit32.bor(result, bit32.lshift(bit32.band(value, 0x7F), i * 7))
			if not bit32.btest(value, 0x80) then
				break
			end
		end

		return result
	end
```